File: src/services/Auth/Auth.API/Models/LoginModel.cs
using System.ComponentModel.DataAnnotations;

namespace Auth.API.Models
{
  public record LoginModel
  {
    [Required(ErrorMessage = "Nome de usuário ou email está vazio")]
    public string? UsernameOrEmail { get; set; }

    [Required(ErrorMessage = "Senha está vazio.")]
    public string? Password { get; set; }

    public override string ToString()
    {
      return $"{nameof(UsernameOrEmail)}: {UsernameOrEmail}, {nameof(Password)}: {Password}";
    }
  }
}

File: src/mobile/lib/app/utils/models/login_model.dart
class LoginModel {
  String? usernameOrEmail;
  String? password;

  LoginModel({this.usernameOrEmail, this.password});

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = <String, dynamic>{};
    data['usernameOrEmail'] = usernameOrEmail;
    data['password'] = password;
    return data;
  }
}
